velox/exec/StreamingAggregation.cpp

```diff
@@ -318,27 +355,32 @@ void StreamingAggregation::initializeNewGroups(size_t numPrevGroups) {
   newGroups.resize(numGroups_ - numPrevGroups);
   std::iota(newGroups.begin(), newGroups.end(), numPrevGroups);
 
-  for (const auto& aggregate : aggregates_) {
+  for (auto i = 0; i < aggregates_.size(); ++i) {
+    const auto& aggregate = aggregates_.at(i);
     if (!aggregate.sortingKeys.empty()) {
       continue;
     }
 
-    aggregate.function->initializeNewGroups(
-        groups_.data(), folly::Range(newGroups.data(), newGroups.size()));
+    if (aggregate.distinct) {
+      distinctAggregations_.at(i)->initializeNewGroups(
+          groups_.data(), newGroups);
+      continue;
+    }
+
+    aggregate.function->initializeNewGroups(groups_.data(), newGroups);
   }
 
   if (sortedAggregations_) {
-    sortedAggregations_->initializeNewGroups(
-        groups_.data(), folly::Range(newGroups.data(), newGroups.size()));
+    sortedAggregations_->initializeNewGroups(groups_.data(), newGroups);
   }
 }
 
 void StreamingAggregation::initializeAggregates(uint32_t numKeys) {
-  for (auto i = 0; i < aggregates_.size(); ++i) {
-    auto& function = aggregates_[i].function;
+  for (auto& aggregate : aggregates_) {
+    auto& function = aggregate.function;
     function->setAllocator(&rows_->stringAllocator());
 
-    const auto rowColumn = rows_->columnAt(numKeys + i);
+    const auto rowColumn = rows_->columnAt(numKeys++);
```

Comment:
modifying input arguments is an anti pattern; let's not do that

Reply:
Fixed